### PR TITLE
Add option to pass `contentSize` and `layoutMeasurement` when calling `scrollTo`

### PR DESCRIPTION
**Pull request by j-piasecki**
### Summary

Adds option to specify `contentSize` and `layoutMeasurement` when calling `scrollTo`. This way it can also be used for components using `VirutalizedLists`, as those fields are used for updating the currently visible window.

Closes https://github.com/callstack/react-native-testing-library/issues/1540

### Test plan

I'm not sure how to test it properly, as entire `userEvent` API seems to be lacking tests at the moment.

### CODE REVIEW

File: src/user-event/scroll/scroll-to.ts [hunk 1]
```diff
@@ -50,31 +63,46 @@ export async function scrollTo(
 
   ensureScrollViewDirection(element, options);
 
+  emitContentSizeChangeEvent(element, options);
```
Comment (mdjastrzebski) [inline src/user-event/scroll/scroll-to.ts:66]: The sequencing of event matches our experiments with RN runtime:
https://github.com/callstack/react-native-testing-library/wiki/ScrollView-Events

**Review by mdjastrzebski — CHANGES_REQUESTED** [review]
The implementation looks good! Pls add some tests to verify it is working correctly on `FlatList`.

File: src/user-event/scroll/scroll-to.ts [hunk 2]
```diff
@@ -50,31 +63,46 @@ export async function scrollTo(
...
   const finalPosition =
     momentumSteps.at(-1) ?? dragSteps.at(-1) ?? initialPosition;
   setElementScrollOffset(element, finalPosition);
 }
 
+function emitContentSizeChangeEvent(
+  element: ReactTestInstance,
+  options: ScrollToOptions
+) {
+  fireEvent(
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply src/user-event/scroll/scroll-to.ts:93]: `dispatchEvent` has a different signature than `fireEvent`. It expects `event` to be its last argument and passes that directly to the handler, while `fireEvent` accepts variadic arguments and spreads them when passing to the handler.

`contentSizeChange` handler expects `width` and `height` as separate arguments: https://github.com/facebook/react-native/blob/d45a01d6d15530e356065704adb869b1cbbb2576/packages/virtualized-lists/Lists/VirtualizedList.js#L1574

Should I also change the signature of `dispatchEvent` to accept variadic arguments, or is `fireEvent` fine in that case?

[the remark this answers is not in the document]
Comment (mdjastrzebski) [reply src/user-event/scroll/scroll-to.ts:93]: Fair point, I'll add some dispatch function for such cases.

**Review by mdjastrzebski — APPROVED** [review]
I've done some tweaks. Ready for merge!

**Comment by mdjastrzebski** [comment]
🎉 This PR has been release in RNTL [v12.4.2](https://github.com/callstack/react-native-testing-library/releases/tag/v12.4.2)!

**Pull request merged**